test/load-generator/boulder-calls.go

```diff
@@ -323,10 +317,25 @@ func completeAuthorization(authz *core.Authorization, s *State, ctx *context) er
 	}
 	authStr := fmt.Sprintf("%s.%s", chalToSolve.Token, base64.RawURLEncoding.EncodeToString(thumbprint))
 
-	// Add the challenge response to the state's test server
-	s.challSrv.AddHTTPOneChallenge(chalToSolve.Token, authStr)
-	// Clean up after we're done
-	defer s.challSrv.DeleteHTTPOneChallenge(chalToSolve.Token)
+	// Add the challenge response to the state's test server and defer a clean-up.
+	switch chalToSolve.Type {
+	case core.ChallengeTypeHTTP01:
+		s.challSrv.AddHTTPOneChallenge(chalToSolve.Token, authStr)
+		defer s.challSrv.DeleteHTTPOneChallenge(chalToSolve.Token)
+	case core.ChallengeTypeDNS01:
+		// Compute the digest of the key authorization
+		h := sha256.New()
```

Comment:
This is probably something to do in the challenge server code rather than here but it seems like all of this hashing could be moved into `AddDNSOneChallenge` and have the signature be `authz.Identifier.Value, authStr`

Reply:
That makes sense. Since it has to be done in a separate repo I created a tracking issue there: https://github.com/letsencrypt/challtestsrv/issues/8

I think its best to handle as a follow-up separate from this PR.